printing/printer.ml

```diff
@@ -635,34 +635,32 @@ let () =
       optwrite = (fun v -> should_print_dependent_evars := v) }
 
 let print_dependent_evars gl sigma seeds =
-  let rec pr_map sigma = function
-  | [] -> mt ()
-  | (evk, evi)::rest ->
-    let sep = if rest <> [] then ", " else "" in
-    (Evar.print evk) ++ str " : " ++ (Termops.pr_existential_key sigma evk) ++ str sep ++
-         (pr_map sigma rest)
-  in
   if !should_print_dependent_evars then
     let evars = Evarutil.gather_dependent_evars sigma seeds in
     let evars_pp = Evar.Map.fold (fun e i s ->
         let e' = pr_internal_existential_key e in
-        let sep = if s = (str "") then "" else ", " in
-        s ++ str sep ++ e' ++
+        s ++ e' ++
             (match i with
-            | None -> str " open"
+            | None -> (str " : ") ++ (Termops.pr_existential_key sigma e) ++ (str " open")
             | Some i ->
               let using = Evar.Set.fold (fun d s ->
-                  let sep = if s = (str "") then "" else " " in
-                  s ++ str sep ++ (pr_internal_existential_key d))
+                  s ++ str " " ++ (pr_internal_existential_key d))
                 i (str "") in
-              str " using " ++ using))
+              str " using" ++ using)
+          ++ str ", ")
       evars (str "")
     in
-    let map_evars = List.filter (fun b -> let (ev, _) = b in Evar.Map.mem ev evars)
-        (Evar.Map.bindings (Evd.undefined_map sigma)) in
+    let evars_current_pp = match gl with
+        | None -> str "no current goal"
+        | Some gl ->
+           let evars_current = Evarutil.gather_dependent_evars sigma [ gl ] in
+           Evar.Map.fold (fun e _ s ->
+               s ++ (pr_internal_existential_key e) ++ (str " "))
+             evars_current (str "current goal: ")
+    in
     cut () ++ cut () ++
-    str "(dependent evars: " ++ evars_pp ++ str ";" ++ cut () ++
-    str "mapping: " ++ (pr_map sigma map_evars) ++ str ")"
```

Comment:
> folded mapping info into the first part, removing other mapping info code

Was this a necessary change or nice to have?  Just to be clear.

Reply:
There are three changes in these two lines, for every one the answer is different:
* removal of `cut ()`: this is a code simplification, neither necessary nor related to the goal of the PR, similar to the simplification of the separators above.
* removal of `pr_map`: Not necessary, but very nice to have. I produce the mapping information in a, IMO, much simpler way, without the pr_map function and the calls to `Evar.Map.bindings` that were introduced in 912527f816b4219a757eb94c00a5a40fbb5ce583. As I said before, I would suggest to squash an updated version of this change with 912527f816b4219a757eb94c00a5a40fbb5ce583, such that `pr_map` will never appear in the master branch.
* addition of `evars_current_pp`: necessary, this produces the output according to requirement 4 of #10420